tests/rptest/tests/compatibility/kafka_streams_test.py

```diff
@@ -550,3 +551,197 @@ def test_kafka_streams_map_function(self):
+
+
+class KafkaStreamsInteractiveQueries(RedpandaTest):
+    """
+    Test KafkaStreams InteractiveQueries app which locates and
+    queries kafka streams state stores. The state stores used in this app are
+    a KeyValue store and a Windowed Store.
+
+    Outside of KStreams, the app does simple prod-cons so the used RP features and systems are:
+    - topic creation
+    - produce/consume on topic
+    - src/v/kafka
+    - src/v/raft
+    - src/v/storage
+    """
+
+    topics = (TopicSpec(name="TextLinesTopic"), )
+
+    def __init__(self, test_context):
+        super(KafkaStreamsInteractiveQueries,
+              self).__init__(test_context=test_context)
+        self._ctx = test_context
+        self._ports = [7070, 7071]
+        self._store_names = ["windowed-word-count", "word-count"]
+
+    def list_all_running_instances(self, res):
+        # Example response looks like
+        # [{'host': 'docker_n_4', 'port': 7070, 'storeNames': ['windowed-word-count']},
+        # {'host': 'docker_n_5', 'port': 7071, 'storeNames': ['word-count']}]
+
+        if len(res) != 2:
+            return False
+
+        if len(res[0]["storeNames"]) != 1 or len(res[1]["storeNames"]) != 1:
+            return False
+
+        valid_host = "docker_n_" in res[0]["host"] and "docker_n_" in res[1][
+            "host"]
+
+        valid_port = res[0]["port"] in self._ports and res[1][
+            "port"] in self._ports
+
+        valid_store_name = res[0]["storeNames"][0] in self._store_names and res[
+            1]["storeNames"][0] in self._store_names
+
+        return valid_host and valid_port and valid_store_name
+
+    def list_instances_by_store_name(self, res):
+        # Example response looks like
+        # [{'host': 'docker_n_5', 'port': 7071, 'storeNames': ['word-count']}]
+
+        if len(res) != 1:
+            return False
+
+        if len(res[0]["storeNames"]) != 1:
+            return False
+
+        valid_host = "docker_n_" in res[0]["host"]
+
+        valid_port = res[0]["port"] in self._ports
+
+        valid_store_name = res[0]["storeNames"][0] in self._store_names
+
+        return valid_host and valid_port and valid_store_name
+
+    def get_instance_for_word_in_store(self, res):
+        # Example response looks like
+        # {'host': 'docker_n_4', 'port': 7070, 'storeNames': ['word-count']}
+
+        if len(res["storeNames"]) != 1:
+            return False
+
+        valid_host = "docker_n_" in res["host"]
+
+        valid_port = res["port"] in self._ports
+
+        valid_store_name = res["storeNames"][0] in self._store_names
+
+        return valid_host and valid_port and valid_store_name
+
+    def get_latest_value_for_word_in_store(self, res):
+        # Example response looks like
+        # {"key":"hello","value":3}
+
+        valid_word = res["key"] in KafkaStreamExamples.QUERIES_HIST.keys()
+
+        valid_count = res["value"] == KafkaStreamExamples.QUERIES_HIST.get(
+            res["key"], False)
```

Comment:
Imagine that key isn't in in the QUERIES_HIST but res["value"] is False; the method should fail but it passes because get defaults to False.  It's better to rewrite it to be precise.

```
if res["key"] not in KafkaStreamExamples.QUERIES_HIST:
    return False
return res["value"] ==KafkaStreamExamples.QUERIES_HIST[res["key"]]
```

Reply:
I agree that you're edit is more readable but the test would still fail in your example because the key is not in QUERIES_HIST (thus valid_word is False) and, as you said, the value defaults to False (thus valid_count is False). Therefore line 643 also returns False (didn't pass). I'll test this out just to make sure.